wgsl/index.bs

```diff
@@ -9040,49 +9042,59 @@ specification for interface validation requirements.
 * Add new shader stages.
 
-## Enable Directive ## {#enable-directive-section}
+There are two kinds of extensions: [=enable-extensions=] and [=software extensions=].
 
-An <dfn noexport>enable directive</dfn> indicates that the functionality
-described by a particular named
-[=extension=] may be used.
-The grammar rules imply that all enable directives [=shader-creation error|must=] appear before any [=declarations=] or [[#const-assert-statement|const assertions]].
+## Enable Extensions ## {#enable-extensions-sec}
 
-The directive uses a [=context-dependent name=] to name the extension.
+An <dfn noexport>enable-extension</dfn> is an [=extension=] whose functionality is available only if:
+* The implementation supports it, and
+* The shader explicitly requests it via an [=enable directive=], and
+* The corresponding WebGPU {{GPUFeatureName}} was one of the required features requested when creating the {{GPUDevice}}.
 
-In particular, an extension name may be spelled the same as a [=keyword=] or [=reserved word=],
-but is not interpreted as any of those.
+[=Enable-extensions=] are intended to embody hardware functionality that is not universally available.
 
-The valid extensions are listed in [[#extension-list]].
+An <dfn noexport>enable directive</dfn> is a [=directive=] that turns on support for an enable-extension.
+A [=shader-creation error=] results if the implementation does not support the extension.
 
 <div class='syntax' noexport='true'>
   <dfn for=syntax>enable_directive</dfn> :
 
-    | <a for=syntax_kw lt=enable>`'enable'`</a> [=syntax/extension_name=] <a for=syntax_sym lt=semicolon>`';'`</a>
+    | <a for=syntax_kw lt=enable>`'enable'`</a> [=syntax/enable_extension_name=] <a for=syntax_sym lt=semicolon>`';'`</a>
 </div>
 
-Note: The grammar rule includes the terminating semicolon token,
-ensuring the additional functionality is usable only after that semicolon.
-Therefore any WGSL implementation can parse the entire `enable` directive.
-When an implementation encounters an enable directive for an unsupported extension,
-the implementation can issue a clear diagnostic.
+Like other directives, an [=enable directive=], if present, must appear before all [=declarations=] and [[#const-assert-statement|const assertions]].
+Extension names are not [=identifiers=]: they do not [=resolves|resolve=] to [=declarations=].
+
+The valid [=enable-extensions=] are listed in the following table.
+<table class='data'>
+  <caption>Enable-extensions</caption>
+  <thead>
+    <tr><th>WGSL enable-extension
+        <th>WebGPU {{GPUFeatureName}}
+        <th>Description
+  </thead>
+  <tr><td><dfn noexport dfn-for="extension">`f16`</dfn>
+      <td>`"shader-f16"`
+      <td>Keyword `f16` and any [=floating point literal=] with a `h` suffix is valid if and only if this extension is enabled. Otherwise, using `f16` keyword or any [=floating point literal=] with a `h` suffix will result in a [=shader-creation error=].
```

Comment:
f16 is no longer a keyword.

```suggestion
      <td>The [=f16=] type is valid to use in the WGSL program. Otherwise, using [=f16=] (directly or indirectly) will result in a [=shader-creation error=].
```